EIPS/eip-7212.md

```diff
@@ -81,13 +81,13 @@ r' == r
 
 ```
 
-### Required Checks in the Verification
+### Required Checks in Verification
 
-Some requirements have to be checked to understand if the given signature components are valid:
+Following requirements **must** be checked by the precompile to verify signature components are valid:
 
-- Verify that both `r` - `s` values are greater than 0 and less than the curve order.
-- Verify that s is equal to or less than half of the order of the subgroup to prevent signature malleability.
-- Verify that the point formed by (`x`, `y`) values is on the curve and both components are in between 0 and the `p` value of the curve.
+- Verify that `r` and `s` values are valid scalar field elements, i.e. they are in `(0, n)` (exclusive) where `n` is the order of the subgroup.
+- Verify that `s <= n // 2`, where `n` is the order of the subgroup to prevent signature malleability.
+- Verify that the point formed by `(x, y)` is on the curve and both `x` and `y` are in `(0, p)` (exclusive) where `p` is the prime field modulus.
```

Comment:
I would actually suggest that these be deleted from this EIP to match compatibility exactly with the NIST specs which does not specify this. In particular, if I understand correctly, at the moment a valid Implementation of P256 signing according to the NIST spec can fail this verifier. This would not be ideal for the purpose of this precompile.

It is anyway relatively straightforward/cheap to add these checks separately in EVM byte code if a contract implementation does want them.